File: app/src/main/java/com/example/mvctutorial/designpattern/_16mediator/ChatMediator.java
package com.example.mvctutorial.designpattern._16mediator;

import com.example.mvctutorial.designpattern._16mediator.content.Colleague;
import com.example.mvctutorial.designpattern._16mediator.content.Mediator;

class ChatMediator extends Mediator {

    @Override
    public void mediate(String data) {
        for (Colleague colleague : colleagues) {
            //중재가능성.
            colleague.handle(data);
        }
    }

}

File: app/src/main/java/com/example/mvctutorial/designpattern/_16mediator/ChatMediator.kt
package com.example.mvctutorial.designpattern._16mediator

import com.example.mvctutorial.designpattern._16mediator.content.Mediator

class ChatMediator: Mediator() {

    override fun mediate(data: String) {
        for (colleague in colleagues) {
            //중재가능성.
            colleague.handle(data)
        }
    }

}
